x-pack/filebeat/input/azureblobstorage/input.go

```diff
@@ -99,30 +99,39 @@ func tryOverrideOrDefault(cfg config, c container) container {
 		}
 		c.MaxWorkers = &maxWorkers
 	}
+
 	if c.Poll == nil {
 		var poll bool
 		if cfg.Poll != nil {
 			poll = *cfg.Poll
 		}
 		c.Poll = &poll
 	}
+
 	if c.PollInterval == nil {
 		interval := time.Second * 300
 		if cfg.PollInterval != nil {
 			interval = *cfg.PollInterval
 		}
 		c.PollInterval = &interval
 	}
+
 	if c.TimeStampEpoch == nil {
 		c.TimeStampEpoch = cfg.TimeStampEpoch
 	}
+
 	if c.ExpandEventListFromField == "" {
 		c.ExpandEventListFromField = cfg.ExpandEventListFromField
 	}
+
 	if len(c.FileSelectors) == 0 && len(cfg.FileSelectors) != 0 {
 		c.FileSelectors = cfg.FileSelectors
 	}
-	c.ReaderConfig = cfg.ReaderConfig
+
+	if isConfigEmpty(c.ReaderConfig) {
+		c.ReaderConfig = cfg.ReaderConfig
+	}
```

Comment:
I think this needs a comment explaining what is going on here; not the action, but the assumptions that are made that make the action OK.